cuckoo/common/mongo.py

```diff
@@ -45,4 +47,161 @@ def connect(self):
+        return self.db.analysis.find({"target.file.sha512": {"$regex": value, "$options": "-i"}}).sort([["_id", -1]])
+
+    def search_process_args(self, value):
+        return self.db.analysis.find({"behavior.processes.command_line": {"$regex": value, "$options": "-i"}}).sort([["_id", -1]])
+
+    def search_regkey_read(self, value):
+        return self.db.analysis.find({"behavior.summary.regkey_read": {"$elemMatch": {"$regex": value, "$options": "-i"}}}).sort([["_id", -1]])
+
+    def search_regkey_opened(self, value):
+        return self.db.analysis.find({"behavior.summary.regkey_opened": {"$elemMatch": {"$regex": value, "$options": "-i"}}}).sort([["_id", -1]])
+
+    def search_regkey_written(self, value):
+        return self.db.analysis.find({"behavior.summary.regkey_written": {"$elemMatch": {"$regex": value, "$options": "-i"}}}).sort([["_id", -1]])
+
+    def extract_database(self, results):
+        # Get data from cuckoo db
+        db = Database()
+        analyses = []
+
+        for result in results:
+            new = db.view_task(result["info"]["id"])
+
+            if not new:
+                continue
+
+            new = new.to_dict()
+
+            if result["info"]["category"] == "file":
+                if new["sample_id"]:
+                    sample = db.view_sample(new["sample_id"])
+                    if sample:
+                        new["sample"] = sample.to_dict()
+
+            analyses.append(new)
+        return analyses
+
+    def search(self, term, value):
+        if not self.enabled:
+            return
+
+        results = []
+        if term == "name":
+            results = self.search_filename(value)
+        elif term == "type":
+            results = self.search_filetype(value)
+        elif term == "string":
+            results = self.search_string(value)
+        elif term == "ssdeep":
+            results = self.search_ssdeep(value)
+        elif term == "crc32":
+            results = self.search_crc32(value)
+        elif term == "file":
+            results = self.search_file(value)
+        elif term == "key":
+            results = self.search_key(value)
+        elif term == "mutex":
+            results = self.search_mutex(value)
+        elif term == "domain":
+            results = self.search_domain(value)
+        elif term == "ip":
+            results = self.search_ip(value)
+        elif term == "signature":
+            results = self.search_signature(value)
+        elif term == "url":
+            results = self.search_url(value)
+        elif term == "imphash":
+            results = self.search_imphash(value)
+        elif term == "args":
+            results = self.search_process_args(value)
+        elif term == "regkey_read":
+            results = self.search_regkey_read(value)
+        elif term == "regkey_opened":
+            results = self.search_regkey_opened(value)
+        elif term == "regkey_written":
+            results = self.search_regkey_written(value)
```

Comment:
All of these `if`-statements could certainly be improved through the usage of a dictionary.